src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalysisContextHelpers.cs

```diff
@@ -84,10 +84,15 @@ internal static void VerifyIOperationFeatureFlag(bool isIOperationFeatureEnabled
 
         private static void VerifyDiagnosticLocationInCompilation(string id, Location location, Compilation compilation)
         {
-            if (location.IsInSource && !compilation.ContainsSyntaxTree(location.SourceTree))
+            if (!location.IsInSource)
+            {
+                return;
+            }
+
+            if (!compilation.ContainsSyntaxTree(location.SourceTree) || location.SourceTree.Length < location.SourceSpan.End)
```

Comment:
@heejaechang See http://source.roslyn.io/#Roslyn.Compilers.CSharp.Semantic.UnitTests/Diagnostics/DiagnosticAnalyzerTests.cs,891 for related tests. You can add a similar one for new check.